### ISSUE TITLE
Yad height setting not work when text is too long

### ISSUE THREAD

**Issue by salpaca** [v1cont/yad#140]
Hello,

I am using yad to display notification in BASH. However, there is some trouble while showing long text with specified height.

My os system verison is Ubuntu **18.04.4** LTS, yad version is **0.40.0** (GTK+ 3.22.30). At first, I try to display a short text with height equal to 100 and everything is fine as follows.

`/usr/bin/yad --text="Set height equal to 100" --width=500 --height=100 --on-top --center --button="OK" --undecorated --borders=15 --image="dialog-warning"`

![image](https://user-images.githubusercontent.com/11244505/128963731-d861c1c8-1b52-47ba-a222-6c2cd5046722.png)

Then, if i try to display a longer text. The shown dialog height is actually larger than specified 100 value **with a lot of space below the string**.

`/usr/bin/yad --text="Set height equal to 100;Set height equal to 100;Set height equal to 100;" --width=500 --height=100 --on-top --center --button="OK" --undecorated --borders=15  --image="dialog-warning"`

![image](https://user-images.githubusercontent.com/11244505/128963810-4fb441e3-cb7a-4eef-b89b-14e836ae4eef.png)

It seems that the dialog default height will increase if text is long enough. It will show default height when setting height is lower than default value. The setting height will work only when the specified value is larger than default height. For example, it will be normal if i set height equal to 500 for previous example2.

`/usr/bin/yad --text="Set height equal to 100;Set height equal to 100;Set height equal to 100;" --width=500 --height=500 --on-top --center --button="OK" --undecorated --borders=15  --image="dialog-warning"`

![image](https://user-images.githubusercontent.com/11244505/128964002-3aac9944-f38c-41d3-8a95-80fc4861a021.png)

I would like to know whether there is one method that makes the specified height always work no matter how long the text string height is? Thanks!

**Comment by frostworx** [comment]
Hi,
would you say this is a duplicate from my https://github.com/v1cont/yad/issues/136?

**Comment by Batwam** [comment]
I agree with this comment. I have seen this issue multiple times. Based on investigations, it seems that this is less prominent for screen with higher resolutions (ie. UHD and above), possibly due to the use of fractional scaling... not too sure. 

When using yad in script code, I am having to adjust the heights by 15-20% between HD and UHD screens to adjust for these oversized dialogs. It looks like yad is trying to calculate the optimal dialog size based on the amount of characters and visibly  overestimating it.

Would someone know in which part of the yad source code this size calculation is made?

**Comment by step-** [comment]
* This issue doesn't happen with version 0.42.x compiled for GTK 2.
* It happens with version 0.42.x compiled for GTK 3.
* It happens with master 12.0 compiled for GTK3.

So I think this issue is caused by the different design of the `GtkLabel` widget in GTK 3 vs GTK 2.
 
```sh
text='jdlska ldksajl alksdjlask jaldkjalskdjalskdjal kjdlaksjdla kjdlakjsd lkasjd lakdjal kjdlaksjd lakjsd lakjdlak jldkajsldk jalskdj laksj dlakdjlakjdlaksjdlaksjdlak jlakdjalksj laksjdalksj dlakjdakjs d jdlaskdjla kjlakjsdl kajs lkdjals kjdlakjs lkdajls kjdlaksj lakjs ldkajsl kjalksjd lakj lakdjals kjalksjd laksjdl aksjdla kjsldkajs ldkajs lakdjal ksjalksdj laksdlak jdlaksjd laksd lakdjlaksdj lakdjalskd jalkdjlakdjalkdjalskdj alk'

yad_gtk2 --text="$text" # gtk2 0.40.x

yad_gtk3 --text="$text" # gtk3 0.40.x

yad_master --text="$text" # gtk3 master 12.0
```

**yad_gtk2**

![yad-20221024](https://user-images.githubusercontent.com/1414102/197523181-86570b8e-69e2-42fe-94d1-1531249ee373.png)

**yad_gtk3**

![yad-20221024-001](https://user-images.githubusercontent.com/1414102/197523441-1e01ed60-1cad-4709-a036-2edac91509ab.png)

**yad_master**

![yad-20221024-002](https://user-images.githubusercontent.com/1414102/197523481-d662ec6c-2dac-4152-a8c0-17bfa3fc0909.png)
